### ISSUE TITLE
"Resolve in (next) release" should not repoen for error in old releases

### ISSUE THREAD

**Issue by jeengbe** [getsentry/sentry#44881]
### Environment

SaaS (https://sentry.io/)

### Version

n/a

### Link

n/a

### DSN

n/a

### Steps to Reproduce

When an error occurs in release A and is set to "Resolve in next release", it is still unresolved with the next error that comes in release A (or prior releases even), triggering a regression notification.

### Expected Result

Instead, if I mark an error to be resolved in the next release, it should be treated as resolved (and not regressed) for releases that are prior to release B. 
If an error tagged with release B raises that same error, only then it should be considered a regression.

### Background

Working with web clients, there will always be cached clients or bots running old versions, triggering long resolved errors and sending those to Sentry. While I have no strong other opinions about how to deal with those, a single event of an issue fixed a month ago should not trigger a regression notification. Currently, this leads to errors like
![image](https://user-images.githubusercontent.com/51749023/220300854-58644e03-cb4a-401a-909a-8bce70e0f112.png)
caused by an ages old client.
```[tasklist]
### Tasks
```

**Comment by snigdhas** [comment]
Thanks for filing this @jeengbe - adding this to our backlog to look into and fix.

**Comment by jeengbe** [comment]
Hi @snigdhas, what's the status on this? This feature is quite a big deal in terms of issue management.

**Comment by snigdhas** [comment]
Hi @jeengbe, the team is unlikely to get to it this quarter. I'll leave this to @brianthi to prioritize.

**Comment by odinho** [comment]
+1 this and adding some details from our (wanted) use of this.

This is a quite weird issue and I've written in to support about it before, where they pointed me to semver, which we already were using. I found out that we should actually make our PR preview releases have an unambigious lower version number after this to make sure we didn't run into it.

Turns out: Sentry doesn't understand semver versions even with this hack. Example:

142.0.0  is newest stable release
142.1.0  we do a new merge with a bugfix, now the version on main
<mark issue as resolved in 142.1.0>
142.0.0-odin-my-cool-pr  I open a new PR based on the tag commit for stable

Now suddenly this PR is the one used by Sentry as 'current release'. And it will send me a regression email about the resolved issue.

Even worse, stable-releases (which people stay on a looong time) has also suddenly been able to mark things as regression. Say `138.0.0` with the above. :)

**Comment by scttcper** [comment]
@odinho @jeengbe are you using what sentry considers to be semantic versions? https://docs.sentry.io/platforms/rust/guides/actix-web/configuration/releases/#bind-the-version 

Basically `142.0.0` isn't enough. It has to have a package name `something@142.0.0`

**Comment by jeengbe** [comment]
We use Git commit hashes as version identifiers.

**Comment by odinho** [comment]
We use `web@142.0.0` etc. Our first try, way back when didn't have this. It was not very easy to figure all of those restrictions out.

**Comment by jeengbe** [comment]
These shouldn't be restrictions, though. Sentry does have information about the order of releases, be they merely commit hashes, which it should be able to determine "next release" with 🙂

**Comment by scttcper** [comment]
@odinho Yeah we've seen quite a few people do that. We're currently working on making it more clear if you're using semver.

@jeengbe With semver we're able to compare version numbers before marking the issue as a regression, commit hash versions we compare using `COALESCE(date_released, date_added)`. So if the release was more recently created than the version the issue was resolved in it can be marked as a regression.

We'll also be making it more clear when an issue is marked as regression that we say how we compared the two releases whether that is semver or date
